specification/network/resource-manager/Microsoft.Network/stable/2019-09-01/bastionHost.json

```diff
@@ -290,6 +262,255 @@
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/BastionShareableLinkRequest"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns a list of shareable links for the specified VMs, giving appropriate error messages as needed.",
+            "schema": {
+              "$ref": "#/definitions/BastionShareableLinkListResult"
+            }
+          },
+          "400": {
+            "description": "Bad request. Check the message for details."
+          },
+          "404": {
+            "description": "No access to the bastion or one of the requested virtual machines."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns the Bastion Shareable Links for the request VMs": {
+            "$ref": "./examples/BastionShareableLinkGet.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts/{bastionHostName}/getactivesessions": {
+      "post": {
+        "tags": [
+          "BastionHosts"
+        ],
+        "operationId": "GetActiveSessions",
+        "description": "Returns the list of currently active sessions on the Bastion.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/BastionHostName"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation returns a list of active sessions on the Bastion.",
+            "schema": {
+              "$ref": "#/definitions/BastionActiveSessionListResult"
+            }
+          },
+          "202": {
+            "description": "Accepted and the operation will complete asynchronously."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Returns a list of currently active sessions on the Bastion": {
+            "$ref": "./examples/BastionSessionsList.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts/{bastionHostName}/disconnectactivesessions": {
```

Comment:
How many active sessions do you see having per bastion host?

I ask because this is effectively a bulk delete operation which is not supported for resources without the use of batch. If the set of activesessions is large enough, modeling these APIs as listActiveSessions and disconnectActiveSessions could be acceptable. Typically this would be modeled as a resource with get - ActiveSession(s) and delete - ActiveSession (doesn't seem you'd need a put as create happens in the dataplane).